doc/whats_new/v0.22.rst

```diff
@@ -427,6 +427,13 @@ Changelog
   a proper error message is raised if X contains some negative entries.
   :pr:`14680` by :user:`Alex Gramfort <agramfort>`.
 
+- |Enhancement| :func:`utils.safe_sparse_dot` works between 3D+ ndarray and
+  sparse matrix. 
+  |Fix| The documented behavior of the ``dense_output`` parameter of
+  :func:`utils.safe_sparse_dot` was not the intended behavior and now matches
+  the actual behavior of the function.
```

Comment:
So the second part is a documentation fix? Not sure it needs a what's new entry?

Reply:
I was not sure about that. It's true that the behavior did not change, only it's documentation. I removed it.